app/controllers/optIn/ChooseYearController.scala

```diff
@@ -57,19 +57,31 @@ class ChooseYearController @Inject()(val optInService: OptInService,
   def show(isAgent: Boolean = false): Action[AnyContent] = auth.authenticatedAction(isAgent) {
     implicit user =>
       withRecover(isAgent) {
-        Future.successful(Ok(view(ChooseTaxYearForm(optInService.availableOptInTaxYear().map(_.toString)), viewModel(isAgent))))
+
+        /* todo to be removed, session data should be setup at start of journey */
```

Comment:
Would it be possible to implement this as the initial fetch of the data, store it in the Journey, and then fetch it back? Then when the start of journey is established, it's just as case of moving the initial fetch & store?

Reply:
done